include/nbl/ext/ToneMapper/CToneMapper.h

```diff
@@ -186,48 +204,36 @@ class CToneMapper : public core::IReferenceCounted, public core::InterfaceUnmova
 			bool usingTemporalAdaptation=false
 		);
 
-		//
-		static inline core::smart_refctd_ptr<video::IGPUImageView> createViewForImage(
-			video::IVideoDriver* driver, bool usedAsInput,
-			core::smart_refctd_ptr<video::IGPUImage>&& image,
-			const asset::IImage::SSubresourceRange& subresource
-		)
-		{
-			if (!driver || !image)
-				return nullptr;
-
-			auto nativeFormat = image->getCreationParameters().format;
-
-			video::IGPUImageView::SCreationParams params;
-			params.flags = static_cast<video::IGPUImageView::E_CREATE_FLAGS>(0u);
-			params.image = std::move(image);
-			params.viewType = video::IGPUImageView::ET_2D_ARRAY;
-			params.format = usedAsInput ? getInputViewFormat(nativeFormat):getOutputViewFormat(nativeFormat);
-			params.components = {};
-			params.subresourceRange = subresource;
-			return driver->createGPUImageView(std::move(params));
-		}
-
 		// we expect user binds correct pipeline, descriptor sets and pushes the push constants by themselves
-		_NBL_STATIC_INLINE_CONSTEXPR uint32_t DEFAULT_WORKGROUP_DIM = 16u;
 		static inline void dispatchHelper(
-			video::IVideoDriver* driver, const video::IGPUImageView* outputView,
-			bool issueDefaultBarrier=true, uint32_t workGroupSizeX=DEFAULT_WORKGROUP_DIM, uint32_t workGroupSizeY=DEFAULT_WORKGROUP_DIM
-		)
+			video::IGPUCommandBuffer* cmdbuf,
+			const video::IGPUImageView* outputView,
+			const asset::E_PIPELINE_STAGE_FLAGS srcStageMask,
+			const uint32_t srcImageBarrierCount,
+			const video::IGPUCommandBuffer::SImageMemoryBarrier* srcImageBarriers,
+			const asset::E_PIPELINE_STAGE_FLAGS dstStageMask,
+			const uint32_t dstImageBarrierCount,
+			const video::IGPUCommandBuffer::SImageMemoryBarrier* dstImageBarriers,
+			const bool usingTemporalAdaptation = false,
+			uint32_t workGroupSizeX=DEFAULT_WORKGROUP_DIM,
+			uint32_t workGroupSizeY=DEFAULT_WORKGROUP_DIM)
```

Comment:
we don't give option of workgroup size when creating the shader, we shouldn't give option here either.